test/functional/index.test.js

```diff
@@ -2,8 +2,8 @@
 var test = require('./shared').assert;
 var setupDatabase = require('./shared').setupDatabase;
 const expect = require('chai').expect;
-const withClient = require('./shared').withClient;
-const withMonitoredClient = require('./shared').withMonitoredClient;
+const shared = require('./shared');
+const { withClient, withMonitoredClient } = shared;
```

Comment:
if we're destructuring anyway it looks like just about all of these come from `shared`:
```
const {
  withClient,
  withMonitoredClient,
  assert,
  setupDatabase
} = require('./shared');
```

Reply:
@mbroadst I updated [a few things](https://github.com/mongodb/node-mongodb-native/compare/d88e609..299927f) since your review, can you take a look when you get a chance? Refactoring `withMonitoredClient` exposed an issue in `withClient`, where it was swallowing errors returned by the `done(err)` callback in tests. I've updated `withClient` to now throw those errors after cleanup.